woocommerce/woocommerce-ios
# [Mobile Payments] Show onboarding in Simple Payments when required

Closes: #6753
- #6753 


### Description


This follows from work on Order Details:
- #6761 
- #6770 

On opening Order Details, we now check the onboarding state for In-Person Payments, and show onboarding if required when collecting payments.

This PR extracts that presentation code, and reuses it in the Simple Payments flow.

Now, we check the onboarding state when the `Payment Method` screen is shown, and show the onboarding flow if required when the user taps `Card`.

In the same way as on Order Details, when the user resolves the onboarding issues, we pop back to Simple Payments and start the payment flow.

#### Note on testing

5ec5aa16e1dc6ca64d8376ad29ab85d8042482ca changes the behaviour of the `Collect Payment` button on order details, and the `Card` payment method row in Simple Payments. In both cases, it makes these UI elements available more often, to make the changes easier to test.

That commit also causes the following `SimplePaymentsMethodsViewModelTests` to fail:
- [test_card_row_is_not_shown_for_non_cpp_store](https://github.com/woocommerce/woocommerce-ios/blob/f7e1d4d17443a201e3bb47a5309d8afe16da31e5/WooCommerce/WooCommerceTests/ViewRelated/Orders/Simple%20Payments/SimplePaymentsMethodsViewModelTests.swift#L305)
- [test_card_row_state_changes_when_store_state_changes](https://github.com/woocommerce/woocommerce-ios/blob/f7e1d4d17443a201e3bb47a5309d8afe16da31e5/WooCommerce/WooCommerceTests/ViewRelated/Orders/Simple%20Payments/SimplePaymentsMethodsViewModelTests.swift#L314)

🚫 I will revert that commit before merging. The above tests will then pass.

### Testing instructions


Using a store with both Stripe and WCPay plugins installed and activated:

1. On the orders tab, tap + and use Simple Payment
2. Set an amount, and tap next to go through to Payment Method
3. Tap Card
4. Observe that you see the "Multiple Conflicting Plugins" onboarding screen
5. Use the Manage Plugins CTA and disable one of the payment plugins
6. Tap Done
7. Observe that you see an onboarding loading screen, which then dismisses and returns to the Simple Payment flow
8. Observe that the Collect Payment flow starts.
9. Connect a reader and collect payment.

Reactivate the plugin you deactivated, and then check the flow from Order Details (as the onboardingViewController pop on completion code has changed slightly.)

1. On the orders tab, tap + and use Create Order
2. Add a product, and tap next to go through to Order Details
3. Tap Collect Payment
4. Observe that you see the "Multiple Conflicting Plugins" onboarding screen
5. Use the Manage Plugins CTA and disable one of the payment plugins
6. Tap Done
7. Observe that you see an onboarding loading screen, which then dismisses and returns to the Order Details
8. Observe that the Collect Payment flow starts.
9. Connect a reader and collect payment.

### Screenshots


https://user-images.githubusercontent.com/2472348/166803754-0b76f5d5-ebe0-40eb-be8c-5209e38b7b0c.mp4



---
